File: yetutil/src/main/java/net/yet/theme/Colors.java
package net.yet.theme;

import android.graphics.Color;

import net.yet.ext.GraphicExtKt;
import net.yet.util.Hex;

public class Colors {
	public static final int BLACK = 0xFF000000;
	public static final int DKGRAY = 0xFF444444;
	public static final int GRAY = 0xFF888888;
	public static final int LTGRAY = 0xFFCCCCCC;
	public static final int WHITE = 0xFFFFFFFF;
	public static final int RED = 0xFFFF0000;
	public static final int GREEN = 0xFF00FF00;
	public static final int BLUE = 0xFF0000FF;
	public static final int YELLOW = 0xFFFFFF00;
	public static final int CYAN = 0xFF00FFFF;
	public static final int MAGENTA = 0xFFFF00FF;
	public static final int TRANSPARENT = 0;

	public static final int TRANS = Color.TRANSPARENT;
	public static int LightGray = Color.LTGRAY;
	public static int GrayMajor = 0xFFDDDDDD;
	public static int Disabled = 0xFFDDDDDD;


	public static int GreenMajor = 0xFF00DD00;
	public static int RedMajor = 0xFFDD0000;

	public static int Theme = 0xFF34C4AA;
	public static int PageGray = 0xFFDDDDDD;
	public static int Title = Color.WHITE;
	public static int Fade = 0xFFFF8800;

	public static int TextColorMajor = 0xFF333333;
	public static int TextColorMid = 0xFF555555;
	public static int TextColorMinor = 0xFF777777;
	public static int TextColor = TextColorMajor;

	public static int EditFocus = 0xFF38C4B0;


	public static int Risk = 0xFFCF2C40;
	public static int Safe = 0xFF199055;
	public static int Progress = 0xFF00C864;

	public static int LineGray = Color.LTGRAY;


	public static class Green {
		public static int Light = 0xFF54792A;
		public static int Dark = 0xFF54792A;
	}

	public static int color(String value) {
		return GraphicExtKt.argb(value);
	}
	public static int rgb(int r, int g, int b) {
		return Color.rgb(r, g, b);
	}

	//0xff8800 --> "#ff8800"
	public static String toStringColor(int color) {
		int a = Color.alpha(color);
		int r = Color.red(color);
		int g = Color.green(color);
		int b = Color.blue(color);
		if (a == 0xff) {
			return "#" + Hex.encode((byte) r) + Hex.encode((byte) g) + Hex.encode((byte) b);
		}
		return "#" + Hex.encode((byte) a) + Hex.encode((byte) r) + Hex.encode((byte) g) + Hex.encode((byte) b);
	}


}

File: yetutil/src/main/java/yet/theme/Colors.kt
package yet.theme

import android.graphics.Color
import yet.ext.argb
import yet.ext.color
import yet.util.Hex

object Colors {
	val BLACK = 0xFF000000.color
	val DKGRAY = 0xFF444444.color
	val GRAY = 0xFF888888.color
	val LTGRAY = 0xFFCCCCCC.color
	val WHITE = 0xFFFFFFFF.color
	val RED = 0xFFFF0000.color
	val GREEN = 0xFF00FF00.color
	val BLUE = 0xFF0000FF.color
	val YELLOW = 0xFFFFFF00.color
	val CYAN = 0xFF00FFFF.color
	val MAGENTA = 0xFFFF00FF.color
	val TRANSPARENT = 0

	val TRANS = Color.TRANSPARENT
	var LightGray = Color.LTGRAY
	var GrayMajor = 0xFFDDDDDD.color
	var Disabled = 0xFFCCCCCC.color
	val BlueMajor = 0xFF4990E2.color


	var GreenMajor = 0xFF5ABA39.color
	var RedMajor = 0xFFD83B31.color
	var OrangeMajor = 0xFFF5A623.color
	var CyanMajor = 0xFF50E3C2.color
	var PinkMajor = 0xFFFD6691.color


	var Warning = 0xFFF3A407.color



	var Theme = 0xFF34C4AA.color
	var PageGray = 0xFFDDDDDD.color
	var Title = Color.WHITE
	var Fade = 0xFFFF8800.color
	var Unselected = 0xFFAAAAAA.color

	var TextColorMajor = 0xFF333333.color
	var TextColorMid = 0xFF555555.color
	var TextColorMinor = 0xFF777777.color
	var TextColor = TextColorMajor
	var TextColorUnselected = 0xFF999999.color

	var EditFocus = 0xFF38C4B0.color


	var Risk = 0xFFCF2C40.color
	var Safe = 0xFF199055.color
	var Progress = 0xFF00C864.color

	var LineGray = Color.LTGRAY


	object Green {
		var Light = 0xFF54792A.color
		var Dark = 0xFF54792A.color
	}

	fun color(value: String): Int {
		return argb(value)
	}

	fun rgb(r: Int, g: Int, b: Int): Int {
		return Color.rgb(r, g, b)
	}

	//0xff8800 --> "#ff8800"
	fun toStringColor(color: Int): String {
		val a = Color.alpha(color)
		val r = Color.red(color)
		val g = Color.green(color)
		val b = Color.blue(color)
		if (a == 0xff) {
			return "#" + Hex.encode(r.toByte()) + Hex.encode(g.toByte()) + Hex.encode(b.toByte())
		}
		return "#" + Hex.encode(a.toByte()) + Hex.encode(r.toByte()) + Hex.encode(g.toByte()) + Hex.encode(b.toByte())
	}


}


